src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/stores/DatagridStore.js

```diff
@@ -46,73 +47,75 @@ export default class DatagridStore {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_SORT_COLUMN].join('.');
 
         userStore.setPersistentSetting(key, value);
     }
 
-    static getSortOrderSetting(resourceKey: string, userSettingsKey: string): string {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_SORT_ORDER].join('.');
+    static getSortOrderSetting(datagridKey: string, userSettingsKey: string): string {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_SORT_ORDER].join('.');
 
         return userStore.getPersistentSetting(key);
     }
 
-    static setSortOrderSetting(resourceKey: string, userSettingsKey: string, value: *) {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_SORT_ORDER].join('.');
+    static setSortOrderSetting(datagridKey: string, userSettingsKey: string, value: *) {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_SORT_ORDER].join('.');
 
         userStore.setPersistentSetting(key, value);
     }
 
-    static getLimitSetting(resourceKey: string, userSettingsKey: string): string {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_LIMIT].join('.');
+    static getLimitSetting(datagridKey: string, userSettingsKey: string): string {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_LIMIT].join('.');
 
         return userStore.getPersistentSetting(key);
     }
 
-    static setLimitSetting(resourceKey: string, userSettingsKey: string, value: *) {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_LIMIT].join('.');
+    static setLimitSetting(datagridKey: string, userSettingsKey: string, value: *) {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_LIMIT].join('.');
 
         userStore.setPersistentSetting(key, value);
     }
 
-    static getSchemaSetting(resourceKey: string, userSettingsKey: string): Array<Object> {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_SCHEMA].join('.');
+    static getSchemaSetting(datagridKey: string, userSettingsKey: string): Array<Object> {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_SCHEMA].join('.');
 
         return userStore.getPersistentSetting(key);
     }
 
-    static setSchemaSetting(resourceKey: string, userSettingsKey: string, value: *) {
-        const key = [USER_SETTING_PREFIX, resourceKey, userSettingsKey, USER_SETTING_SCHEMA].join('.');
+    static setSchemaSetting(datagridKey: string, userSettingsKey: string, value: *) {
+        const key = [USER_SETTING_PREFIX, datagridKey, userSettingsKey, USER_SETTING_SCHEMA].join('.');
         userStore.setPersistentSetting(key, value);
     }
 
     constructor(
         resourceKey: string,
+        datagridKey: string,
```

Comment:
Not sure about this change... Has quite an impact, and would theoretical make the `userSettingsKey` obsolete. The `datagridKey` would be good match to remember the user settings, because if the field descriptors differ, it would allow to remember them correctly. However, it would also mean that if you want to have a different user setting for something, you would also have to copy all the content from the datagrid xml configuration...